funcframework/framework_test.go

```diff
@@ -117,10 +117,179 @@ type customStruct struct {
 	Name string `json:"name"`
 }
 
+type testStruct struct {
+	Age  int
+	Name string
+}
+
 type eventData struct {
 	Data string `json:"data"`
 }
 
+func TestRegisterTypedFunction(t *testing.T) {
```

Comment:
Lets add a test case where function input is a struct T1 and return type is a struct T2 where T1 != T2